pandas/core/categorical.py

```diff
@@ -824,7 +825,12 @@ def rename_categories(self, new_categories, inplace=False):
         """
         inplace = validate_bool_kwarg(inplace, 'inplace')
         cat = self if inplace else self.copy()
-        cat.categories = new_categories
+
+        if is_dict_like(new_categories):
+            cat.categories = [new_categories.get(item, item)
```

Comment:
actually make sure the doc string is updated here as well (maybe add a versionchanged tag)